### PR TITLE
Add JSON un/marshaling to RLE, use it to un/marshal BitFields

### PR DESCRIPTION
**Pull request by arajasek**
Motivation: https://github.com/filecoin-project/lotus/pull/2050#pullrequestreview-432214471

### CODE REVIEW

File: rle/rleplus.go [hunk 1]
```diff
@@ -59,3 +60,55 @@ func (rle *RLE) Count() (uint64, error) {
...
+		return err
+	}
+
+	rle.runs = []Run{}
+	val := false
+	for _, v := range buf {
+		rle.runs = append(rle.runs, Run{
+			Val: val,
+			Len: v,
+		})
```
Comment (magik6k) [inline rle/rleplus.go]: make sure that this never adds a 0-length runs

Comment (arajasek) [reply rle/rleplus.go]: Done, thanks (exception made for the first run, since that can be of length zero).

**Comment by Kubuxu** [comment]
We should not have 0 length run at all, even at the start.

**Comment by Kubuxu** [comment]
As in we should not add the initital run if it has length 0.

**Comment by arajasek** [comment]
@Kubuxu Yup, we don't add an initial run length of 0, the "exception" is just that we don't fail on a zero-length run *if* it's the first run.

**Review by Kubuxu — APPROVED** [review]

**Pull request merged**